src/products/services/products.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { ProductEntity } from '@products/entities/product.entity';
+import { GetAllProductsException } from '@products/exceptions/get-all-products.exception';
+
+@Injectable()
+export class ProductsService {
+  private readonly logger: Logger = new Logger(ProductsService.name);
+
+  constructor(
+    @InjectRepository(ProductEntity)
+    private productRepository: Repository<ProductEntity>,
```

Comment:
I would recommend you creating a repository layer in order to separate the database logic from the bussiness.

Example

```node
@EntityRepository(Product)
export class ProductRepository extends Repository<Product> {
// your custom db operations here
}
```
And in your service

```node
 @InjectRepository(CategoryRepository) private categoriesRepo: CategoryRepository
```

So now you have your database logic in your "category.repository.ts" file and the service only calls it without dealing with ORM/SQL functions.

Reply:
Well, yes, I can create the repository layer but I think I can use the injection right now because I'm not using something complex or special query. I can simple use the default methods from `Repository<ProductEntity>`
Or what do you think? Is not a problem to create the repository also

Reply:
Done!